tests/topotests/lib/common_config.py

```diff
@@ -560,6 +567,157 @@ def write_test_footer(tc_name):
     logger.info("="*(len(tc_name)+count))
 
 
+def interface_status(tgen, topo, input_dict):
+    """
+    Delete ip route maps from device
+
+    * `tgen`  : Topogen object
+    * `topo`  : json file data
+    * `input_dict` :  for which router, route map has to be deleted
+
+    Usage
+    -----
+    input_dict = {
+        "r3": {
+            "interface_list": ['eth1-r1-r2', 'eth2-r1-r3'],
+            "status": "down"
+        }
+    }
+    Returns
+    -------
+    errormsg(str) or True
+    """
+    logger.debug("Entering lib API: interface_status()")
+
+    try:
+        global frr_cfg
+        for router in input_dict.keys():
+
+            interface_list = input_dict[router]['interface_list']
+            status = input_dict[router].setdefault('status', 'up')
+            for intf in interface_list:
+                rnode = tgen.routers()[router]
+                interface_set_status(rnode, intf, status)
+
+            # Load config to router
+            load_config_to_router(tgen, router)
+
+    except Exception as e:
+        # handle any exception
+        logger.error("Error %s occured. Arguments %s.", e.message, e.args)
+
+        # Traceback
+        errormsg = traceback.format_exc()
+        logger.error(errormsg)
+        return errormsg
+
+    logger.debug("Exiting lib API: interface_status()")
+    return True
+
+
+def retry(attempts=3, wait=2, return_is_str=True, initial_wait=0):
```

Comment:
This is a really interesting idea, well done!

Reply:
Thanks, i did this as retry is required for multiple verification APIs